### ISSUE TITLE
amp-ad: Ads do not display on Internet Explorer 11

### ISSUE THREAD

**Issue by jamescridland** [ampproject/amphtml#8210]
Google AdSense AMP ads do not display on Internet Explorer 11. The "AD" placeholder continues to show.

However, Google AdSense AMP ads display on Chrome and Firefox; non-AMP ads display fine on Internet Explorer 11. This leads me to suspect that there is no adblocking software on the system.

## How do we reproduce the issue?

1. Visit https://media.info/radio/stations/radio-x-uk in a browser
2. In Chrome (Windows/OSX/Chrome OS), Safari (OSX), and Firefox (Windows) you'll see an ad displayed just above "Facts And Stats".
3. However, in Internet Explorer 11, the ad attempts to load but remains on the "AD" placeholder.

https://media.info/radio/stations/radio-x-uk/listen is a non-AMP page using the same infrastructure. Ads display correctly, including Internet Explorer 11. Given this, I do not believe that this machine is running an ad-blocker. Internet Explorer 11 (given my limited experience of it) does not appear to have any addons or extensions running.

Here's what's happening on Firefox (back left); Chrome (back right), and Internet Explorer 11 (front center). I'm using a library computer, since I don't normally have access to a Windows machine.

![amp-ad-oh-dear](https://cloud.githubusercontent.com/assets/231941/24024873/61a16a1a-0b02-11e7-9a43-4a4c8632a874.jpg)

I've attempted to find other examples of Google AdSense AMP ads to check: but this is a bit difficult. Some ads on The Guardian aren't loading, but they appear to use a different placeholder. The BBC's AMP setup doesn't display ads.

## What browsers are affected?

Internet Explorer 11 (on Windows 7 Enterprise)

## Which AMP version is affected?

AMP version 1489614597532

**Comment by aghassemi** [comment]
/to @zhouyx Setting to P1 until we at least know if this is a regression or not.

**Comment by zhouyx** [comment]
I tested the page on IE. 

> However, in Internet Explorer 11, the ad attempts to load but remains on the "AD" placeholder.

This is not exactly the case. As I found the top loading indicator actually disappears and the ad iframe has been free up. 

However I did find that what we display is the placeholder not the fallback in this case. Which means it could be either `#layoutCallback` fail for the <amp-ad> element, or the <amp-ad> has been unlaid out.

**Comment by zhouyx** [comment]
Investigate into this issue further. 
In the provided example page, `<amp-ad>` doesn't have a `width` value provided where it should [(see doc)](https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/amp-ad.md#behavior). 
When we create an iframe for ad, we depend on this `width` value to set iframe width. [code](https://github.com/ampproject/amphtml/blob/master/src/3p-frame.js#L100). As far as I found setting `iframe.width = undefined` returns an Error (Invalid argument) in IE. Because of this, `#layoutCallback` of the `<amp-ad>` fails in IE. Thus, we remove the loading indicator and keep the placeholder in the example page.

@jamescridland TL;DR Please follow the [doc](https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/amp-ad.md#behavior) and specify width for the ad.

**Comment by jamescridland** [comment]
@zhouyx Thank you for your response. I'm grateful for your investigation. However, the [documents published by AdSense](https://support.google.com/adsense/answer/7183212?hl=en) do not have a width specified in the 'above the fold' section. The docs themselves say:

> For best results, fix the height of the ad unit as 100px and leave the width as responsive so it fits naturally with the rest of your page layout.

I've followed the code specified by AdSense themselves. If this code is incorrect, are you able to escalate this to the AdSense team? I'm unclear of the relationship between amp-ad and adsense.

**Comment by jamescridland** [comment]
Further to the above, @zhouyx ...

The [amp-ad docs](https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/amp-ad.md#behavior) say:

>Supported layouts: fill, fixed, fixed-height, flex-item, nodisplay, responsive

In the [docs for layout](https://www.ampproject.org/docs/guides/author-develop/responsive/control_layout#the-layout-attribute), it says:

> The width attribute must not be present or must be equal to auto.

In short, it appears that I have (quite by accident) followed the docs correctly. Could I ask you to confirm that amp-ad supports fixed-height layout, as the docs currently claim, and if so, please re-open this bug?

**Comment by zhouyx** [comment]
@jamescridland you're right. `amp-ad` support all these layout types. 
Reopen the issue. We need to check for undefined width for IE to fix the issue.

**Comment by zhouyx** [comment]
have #8497 to rephrase the readme of amp-ad.
